internal/provisioner/cluster.go

```diff
@@ -84,30 +87,49 @@ func CreateCluster(provider, size string, logger log.FieldLogger) error {
 func DeleteCluster(clusterId string, logger log.FieldLogger) error {
 	logger = logger.WithField("cluster", clusterId)
 
-	logger.Info("deleting cluster")
-
 	s3StateStore := "dev.cloud.mattermost.com"
 	dns := fmt.Sprintf("%s-kops.k8s.local", clusterId)
 
-	// Temporarily look for the kops output directory as a local folder named `tmp`. See above.
-	outputDir := "tmp"
+	// Temporarily look for the kops output directory as a local folder named after
+	// the cluster ID. See above.
+	outputDir := fmt.Sprintf("cluster-%s", clusterId)
+
+	// Validate the provided cluster ID before we alter state in any way.
+	logger.Info("verifying cluster ID")
+	_, err := os.Stat(outputDir)
+	if err != nil {
+		return fmt.Errorf("failed to find cluster directory %q: %s", outputDir, err)
+	}
 
 	terraform := terraform.New(outputDir, logger)
 	defer terraform.Close()
-	err := terraform.Init()
+	out, err := terraform.Output("cluster_name")
 	if err != nil {
 		return err
 	}
-	err = terraform.Destroy()
-	if err != nil {
-		return err
+	if out != dns {
```

Comment:
Can you elaborate on the thinking here? It seems like we're just checking the `clusterId` in a roundabout way, seeing as we build the DNS up from the `clusterId` anyway. Presumably we'd do the same thing when we switch to S3. Is the concern that the folder might be renamed?

Reply:
Yep! That is the concern. Basically, I want both kops and terraform to tell me that the cluster ID is what we are targeting. This may not be necessary in the future, but with our current local-file setup, I don't want to rely on the directory name being correct.